src/client/method-status.c

```diff
@@ -266,7 +273,21 @@ static int get_status_unit_on(Client *client, char *node_name, char *unit_name,
         return 0;
 }
 
-static int method_status_unit_on(Client *client, char *node_name, char **units, size_t units_count) {
+
+static int on_unit_status_changed(UNUSED sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
+        units_status *user_data = (units_status *) userdata;
+        printf("\033[2J");
+        printf("\033[%d;%dH", 0, 0);
+        print_info_header(*user_data->max_name_len);
+        get_status_unit_on(
+                        user_data->client, user_data->node_name, user_data->unit_name, *user_data->max_name_len);
```

Comment:
You don't need the `struct unit_status`. The node and unit name are in the signal message m anyway and since its only one unit the max len is the length of that string. Passing only the `Client* client` in the userdata is sufficient, for example:

```suggestion
        Client *client = (Client *) userdata;
        clear_screen();
               
        /*
        read information from signal message m
        - node_name
        - unit_name 
        the other information (state and substate can be omitted since we fetch it anyway again)
        */ 
        
        // there is only one unit, so max_name_len is the length of the unit name
        size_t max_name_len = strlen(unit_name);        
        print_info_header(max_name_len);
        get_status_unit_on( node_name, unit_name, max_name_len);
```

__Note: __ 
By removing the `unit_status` struct the memory management becomes easier. At the moment, there is a memory leak due to the `malloc0(sizeof(units_status))` not being freed.